src/test/java/com/teragrep/aer_02/EventDataConsumerTlsTest.java

```diff
@@ -0,0 +1,246 @@
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.aer_02;
+
+import com.codahale.metrics.MetricRegistry;
+import com.teragrep.aer_02.config.source.EnvironmentSource;
+import com.teragrep.aer_02.fakes.PartitionContextFake;
+import com.teragrep.aer_02.fakes.SystemPropsFake;
+import com.teragrep.net_01.channel.socket.TLSFactory;
+import com.teragrep.net_01.eventloop.EventLoop;
+import com.teragrep.net_01.eventloop.EventLoopFactory;
+import com.teragrep.net_01.server.Server;
+import com.teragrep.net_01.server.ServerFactory;
+import com.teragrep.rlo_06.RFC5424Frame;
+import com.teragrep.rlp_01.SSLContextFactory;
+import com.teragrep.rlp_01.client.SSLContextSupplier;
+import com.teragrep.rlp_03.frame.FrameDelegationClockFactory;
+import com.teragrep.rlp_03.frame.delegate.DefaultFrameDelegate;
+import com.teragrep.rlp_03.frame.delegate.FrameContext;
+import com.teragrep.rlp_03.frame.delegate.FrameDelegate;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.TrustManagerFactory;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class EventDataConsumerTlsTest {
+
+    private Server server;
+    private EventLoop eventLoop;
+    private Thread eventLoopThread;
+    private ExecutorService executorService;
+    private final List<String> messages = new ArrayList<>();
```

Comment:
please use concurrent object or internally synchronized one instead of using public synchronized void accept. using non-internally synchronized object is not thread-safe even though two different methods accessing it are synchronized because they are not synchronized on the same lock and therefore there is no happens-before guarantee.

Reply:
added synchronized list